tokio/src/sync/mpsc/bounded.rs

```diff
@@ -150,6 +152,44 @@ impl<T> Receiver<T> {
         poll_fn(|cx| self.poll_recv(cx)).await
     }
 
+    /// Blocking receive to call outside of asynchronous contexts
+    ///
+    /// ```
+    /// use std::thread;
+    /// use tokio::runtime::Runtime;
+    /// use tokio::sync::mpsc;
+    ///
+    /// fn main() {
+    ///     let (mut tx, mut rx) = mpsc::channel::<u8>(10);
+    ///     
+    ///     let sync_code = thread::spawn(move || {
+    ///         assert_eq!(Ok(Some(10)), rx.blocking_recv());
+    ///     });
+    ///
+    ///     Runtime::new()
+    ///         .unwrap()
+    ///         .block_on(async move {
+    ///             let _ = tx.send(10).await;
+    ///         });
+    ///     sync_code.join().unwrap()
+    /// }
+    /// ```
+    pub fn blocking_recv(&mut self) -> Result<Option<T>, crate::park::ParkError> {
+        if Handle::try_current().is_ok() {
+            panic!("blocking_recv cannot be called from within a runtime");
+        }
```

Comment:
This is not the way we should check this. It should not panic in `spawn_blocking` or `block_in_place`.

Reply:
What is the best way to check it? I'm not fully sure from looking at `src/runtime/enter.rs` (I might be looking in the wrong place). 

> It is possible that you will not conditional compilation to turn off the panic when only sync is enabled.

I'm not sure I follow this what you mean by this? It shouldn't be possible to conditionally remove the panic based on the features?